src/chain/auth.sw

```diff
@@ -1,28 +1,39 @@
 library auth;
+//! Functionality for determining who is calling an ABI method
 
-// this can be a generic option when options land
-enum Caller {
-    Some: b256,
-    None: (),
+use ::result::Result;
+use ::address::Address;
+use ::contract_id::ContractId;
+
+pub enum AuthError {
+    ContextError: (),
```

Comment:
I'm thinking that we will likely add another error variant here like `ECRecoverError` in a follow-up PR.